src/dotnet/ReSharperPlugin.FluentAssertions/Helpers/NUnit/NUnitAssertMigrationServiceBase.cs

```diff
@@ -93,6 +104,26 @@ public string GetTextMessage(IInvocationExpression invocationExpression)
             return string.Format(MessageTemplate, nUnitAssert, expression.GetText());
         }
 
+        /// <summary>
+        /// Get actual value for generate equivalent expression
+        /// </summary>
+        /// <param name="arguments">Assert invocation expression arguments</param>
+        /// <returns>Actual argument expression</returns>
+        protected virtual ICSharpExpression GetActualValue(IEnumerable<ICSharpExpression> arguments)
```

Comment:
Maybe we should introduce virtual methods to override the actual NUnit expression rather than only the arguments? That would give us the flexibility to refactor statements like `Assert.Equal(13, myList.Count())` into `myList.Should().HaveCount(13)`

Reply:
I'm not sure I fully understood what you mean, I would love to look in terms of implementation in the code

Reply:
I'm not able to provide an implementation to you since I'm not familiar enough to the R# SDK :( but I can give you some pseudo code:
```c#
// within derivations of NUnitAssertMigrationServiceBase
public class AssertEqualsToHaveCountMigration : NUnitAssertMigrationServiceBase
{
    protected override int NumberOfExistingOperands => 3;
    protected override string ExistingAssertionExpression => "Assert.Equal($0, $1.Count(), $2)";
    protected override string FluentAssertionExpression => "$1.Should().HaveCount($0, $2)";
}

// NUnitAssertMigrationServiceBase
public class NUnitAssertMigrationServiceBase
{
    protected abstract int NumberOfExistingOperands { get; }
    protected abstract string ExistingAssertionExpression { get; }
    protected abstract string FluentAssertionExpression { get; }

    public IExpression CreateMigrationExpression(IInvocationExpression invocationExpression)
    {
        // ...
        var operands = new List<object>();
        for (var i = 0; i < NumberOfExistingOperands; i++)
            operands.Add(arguments.GetValue(i));

        var shouldExpression = factory.CreateReferenceExpression(FluentAssertionExpression, operands.ToArray());
    }

    public bool CanMigrate(IInvocationExpression invocationExpression)
    {
        // Here I don't know the R# SDK syntax, but I'd expect something like this:
        // return invocationExpression.Matches(ExistingAssertionExpression);
    }
}
```